### PR TITLE
Add support for azuredevops_branch_policy_merge_types

### LINKED ISSUE #300: Support branch policy merge types
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for restricting the merge types available for a branch:

* Squash
* Rebase
* Basic (no ff)
* Rebase with merge

### New or Affected Resource(s)

* Resource: `azuredevops_branchpolicy_merge_types`

### Potential Terraform Configuration

```hcl
resource "azuredevops_branch_policy_merge_types" "p" {
  project_id = azuredevops_project.p.id

  enabled  = true
  blocking = true

  settings {
    allow_squash = true
    allow_rebase_and_fast_forward = true
    allow_basic_no_fast_forward = true
    allow_rebase_with_merge = true

    scope {
      repository_id  = azuredevops_git_repository.p.id
      repository_ref = "refs/heads/main"
      match_type     = "Exact"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Add support for restricting branch policy merge types

Issue Number: fixes #300 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+				ValidateFunc: validation.IntAtLeast(1),
+			}
+		}
+		if conflicts, ok := metaField.Field(i).Tag.Lookup("ConflictsWith"); ok {
+			if _, ok := settingsSchema[conflicts]; ok {
+				settingsSchema[conflicts].ConflictsWith = []string{SchemaSettings + ".0." + tfName}
+				settingsSchema[tfName].ConflictsWith = []string{SchemaSettings + ".0." + conflicts}
+			}
+		}
+	}
```
Comment (xuzhang3) [inline azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go]: These codes are redundant, not used in this resource and can be removed.

File: azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	tipe := reflect.TypeOf(policySettings)
+	for i := 0; i < tipe.NumField(); i++ {
+		tfName := tipe.Field(i).Tag.Get("tf")
+		ps := reflect.ValueOf(policySettings)
+		if tipe.Field(i).Type == reflect.TypeOf(true) {
+			settings[tfName] = ps.Field(i).Bool()
+		}
+		if tipe.Field(i).Type == reflect.TypeOf(0) {
+			settings[tfName] = ps.Field(i).Int()
+		}
```
Comment (xuzhang3) [inline azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go]: Unused code and can be removed.

File: azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		tfName := tags.Get("tf")
+		if _, ok := policySettings[apiName]; ok {
+			continue
+		}
+		if tipe.Field(i).Type == reflect.TypeOf(true) {
+			policySettings[apiName] = settings[tfName].(bool)
+		}
+		if tipe.Field(i).Type == reflect.TypeOf(0) {
+			policySettings[apiName] = settings[tfName].(int)
+		}
```
Comment (xuzhang3) [inline azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go]: Unused code and can be removed.

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
Hi @matt-FFFFFF   
Few changes requested but otherwise looks good. You need add the document link to  https://github.com/microsoft/terraform-provider-azuredevops/blob/master/website/azuredevops.erb

File: azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go (hunk 1 above)
Comment (matt-FFFFFF) [reply azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go]: Removed, thanks

File: azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go (hunk 2 above)
Comment (matt-FFFFFF) [reply azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go]: Removed, thanks

File: azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go (hunk 3 above)
Comment (matt-FFFFFF) [reply azuredevops/internal/service/policy/resource_branchpolicy_merge_types.go]: Removed, thanks

**Comment by matt-FFFFFF** [comment]
Hi @xuzhang3 - changes made, thanks for the review

**Comment by matt-FFFFFF** [comment]
Hi @xuzhang3 could you re-run the checks please? They were cancelled for some reason. Thanks :)

**Review by xuzhang3 — APPROVED** [review]
LGTM

**Pull request merged**